doc/tokens.md

```diff
@@ -0,0 +1,137 @@
+
+On a high level, these interactions with the VSS tree need to be considered
+
+ * setting ("writing") signals: For existing datapoints, update the value, specifically this may be 
+   * Updating the _current_value_ of a specific signal, e.g. a provider updating Vehicle speed 
+   * Updating the _target_value_ of a an actuator signal, e.g. an application trying to open a vehicle's trunk by writing to the corresponding actuator
+ * getting ("reading") signals: For existing datapoints, getting or subscribing the value, specifically this may be
+   * Requesting the _current_value of a specific singal, e.g. an application subscribing to Vehicle Speed
+   * Requesting the _target_value_ of an actuator signal, e.g. a provider subscribing to the _target_value_ indicating the intent to open or closing the trunk, in order to effect the change
+ * Modifying the VSS tree itself. This may not be needed for all deployments, but for a typical exampe, consider adding a specfic trailer or implement to a vehicle, where some software components might want to register new datapoints under `Vehicle.Trailer` or `Vehicle.Harvester`.
+
+
+ To describe the level of access for a given client KUKSA.val uses JSON Web Tokens (JWT) according to [RFC7519](https://www.rfc-editor.org/rfc/rfc7519).
+
+ An example token may look like this
+
+ ```json
+{
+  "sub": "kuksa.val",
+  "iss": "Eclipse KUKSA Dev",
+  "iat": 1516239022,
+  "exp": 1606239022,
+  "vss-read":  [
+    "Vehicle.Speed"
+  ]
+}
+``` 
+
+The authorisation is handled by `vss-*` claims. In KUKSA.val we only handle explicit access rights, i.e. a token directly encodes what the owner of said token can do. 
+
+As KUKSA.val is a rather low level component, we are not making assumptions about the security model around your application. Depending on your deployment, you maybe have Role-baed access models, or you work with bearer tokens. You need to make sure, that such access rights are rolled out into explicit descriptions by an IAM/IDM or some other form of policy database in your system.
+
+
+## vss claims 
+
+The following claims are supported.
+
+### vss-read
+This implies a client is able to read (get/subscribe) the current value for sensors, actuators and attributes, as well as the target value for actuators.
+
+**Discussion: Might also be split into read-sensor, read-target? Should this also imply read-meta**
+
+### vss-provide-sensor:
+This implies a client is able to set the current value for sensors and actuators.
+
+**Discussion: Might alternatively be called just "provide", which would fit providers that are "feeders", but  currently in my mind a "provider" could also be of the service variety, and be a component that reacts on updated target values. Might also be jsut "write-current" which may be more obvious if you ar enot into VSS/KUKSA lingo**
```

Comment:
I think it's confusing with the name `vss-provide-sensor` if it includes permissions to provide attributes and actuators as well. `vss-provide` would make more sense.

Not convinced that `write-current` fits into any consistent naming scheme either.

If there is a need for more granular permissions, again I think it would be more workable to add an additional syntax where the individual fields are referenced along with access mode. But this should be in addition to the (probably) most common way of granting "provider rights" for certain paths regardless of whether they are attributes/sensors/actuators.